pkg/promtail/client/client.go

```diff
@@ -85,9 +87,19 @@ func New(cfg Config, logger log.Logger) Client {
 
 		externalLabels: cfg.ExternalLabels,
 	}
+
+	var err error
+	c.client, err = config.NewClientFromConfig(cfg.Client, "promtail")
+	if err != nil {
+		level.Error(c.logger).Log("msg", "error while creating http client", "error", err) //nolint
```

Comment:
We tend not to log in "leaf" functions - the error will get logger at the top of the call tree, ie in the `main` func.

You should do an `error.Wrap` if you want to add more context.